integration/src/main/java/scripts/common/ApiHelpers.java

```diff
@@ -0,0 +1,43 @@
+package scripts.common;
+
+import bio.terra.datarepo.api.JobsApi;
+import bio.terra.datarepo.api.SnapshotsApi;
+import bio.terra.datarepo.client.ApiClient;
+import bio.terra.datarepo.client.ApiException;
+import bio.terra.datarepo.model.JobModel;
+import bio.terra.datarepo.model.SnapshotRequestModel;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class ApiHelpers {
+  public static Map<Object, Object> waitForJobSuccess(ApiClient datarepoClient, String jobId)
+      throws ApiException, InterruptedException {
+    var jobApi = new JobsApi(datarepoClient);
+    JobModel job;
+    do {
+      job = jobApi.retrieveJob(jobId);
+      TimeUnit.SECONDS.sleep(5);
+    } while (job.getJobStatus() == JobModel.JobStatusEnum.RUNNING);
+
+    if (job.getJobStatus() != JobModel.JobStatusEnum.SUCCEEDED) {
+      throw new RuntimeException("Job failed: " + job);
+    }
+    return (Map<Object, Object>) jobApi.retrieveJobResult(jobId);
+  }
+
+  public static UUID synchronousCreateSnapshot(
+      SnapshotsApi snapshotsApi, SnapshotRequestModel request)
```

Comment:
With this code refactored, you could simplify the caller even more by only passing in the field of `request` that are different for each call. I think that comes down to only `profileID` (or make it accept `Dataset`).

Also IMO I would remove a prefix like `synchronous` from these names unless we have an async version and need to disambiguate.

Reply:
the async version would just be regular createSnapshot

Reply:
Maybe, but we don't have an async version, and don't have a need for one in the future. So all `createSnapshot()` method calls in our code are sync. So there's no need to disambiguate sync from async in the method name.